cupyx/cutensor.pyx

```diff
@@ -243,7 +243,7 @@ cpdef TensorDescriptor create_tensor_descriptor(_ndarray_base a):
     """
     handle = _get_handle()
     key = (handle.ptr, a.dtype, tuple(a.shape), tuple(a.strides))
```

Comment:
If we want to relax the alignment requirement, we need to add it as part of the cache key. For different alignments, the tensor descriptor cannot be reused.

Reply:
I got it, I will change that

Reply:
I found that we only need to change the key to cache descripters. For other cached object, desc.ptr is used in keys thus naturally be recreated.

Reply:
I see, it's a good observation, so we're using the tensor descriptor pointer as a proxy to propagate the alignment requirement. As long as we don't have dangling pointers, this should work fine.